### PR TITLE
Fix issue #63, build debuguy with gulp and remove deps of browserfy and grunt

### PR DESCRIPTION
**Pull request by dwi2**
1. Remove Grunt and its' dependency
2. Remove Browserfy and its' dependency
3. Rewrite build script with Gulp
4. Put a pre-commit hook after `npm install`. The hook runs `gulp test` before every commit.

### CODE REVIEW

**Comment by dwi2** [comment]
麻煩 @shamenchens 或 @begeeben review一下 :bow:

File: .pre-commit [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/sh
+
+gulp unit-test
```
Comment (shamenchens) [inline .pre-commit]: should we add jshint to pre-commit hook to prevent committing any unlinted code?

**Comment by shamenchens** [comment]
Looks good to me!

File: .pre-commit (hunk 1 above)
Comment (dwi2) [inline .pre-commit]: `gulp unit-test` depends on `jshint`, so it will run jshint before running `unit-test`

**Pull request merged**